modular_skyrat/modules/medical/code/smartdarts.dm

```diff
@@ -0,0 +1,101 @@
+//The smartdarts themselves
+/obj/item/reagent_containers/syringe/smartdart
+	name = "medical smartdart"
+	desc = "Allows the user to safely inject chemicals at a range without harming the patient."
+	volume = 10
+	icon = 'modular_skyrat/modules/medical/icons/obj/smartdarts.dmi'
+	icon_state = "dart_0"
+	base_icon_state = "dart"
+
+//Code that handles the base interactions involving smartdarts
+/obj/item/reagent_containers/syringe/smartdart/afterattack(atom/target, mob/user, proximity)
+	to_chat(user, span_warning("The [src] is unable to directly inject chemicals."))
+	return
+//A majority of this code is from the original syringes.dm file.
+/obj/item/reagent_containers/syringe/smartdart/afterattack_secondary(atom/target, mob/user, proximity_flag, click_parameters)
+	if(!try_syringe(target, user, proximity_flag))
+		return SECONDARY_ATTACK_CONTINUE_CHAIN
+	if(reagents.total_volume >= reagents.maximum_volume)
+		to_chat(user, span_notice("[src] is full."))
+		return SECONDARY_ATTACK_CONTINUE_CHAIN
+	if(isliving(target))
+		to_chat(user, span_warning("The [src] is unable to take blood."))
+		return SECONDARY_ATTACK_CONTINUE_CHAIN
+	if(!target.reagents.total_volume)
+		to_chat(user, span_warning("[target] is empty!"))
+		return SECONDARY_ATTACK_CONTINUE_CHAIN
+	if(!target.is_drawable(user))
+		to_chat(user, span_warning("You cannot directly remove reagents from [target]!"))
+		return SECONDARY_ATTACK_CONTINUE_CHAIN
+	var/trans = target.reagents.trans_to(src, amount_per_transfer_from_this, transfered_by = user) // transfer from, transfer to - who cares?
+	to_chat(user, span_notice("You fill [src] with [trans] units of the solution. It now contains [reagents.total_volume] units."))
+
+	return SECONDARY_ATTACK_CONTINUE_CHAIN
```

Comment:
First off, add a line between your procs, otherwise it's much less readable.

Secondly, for what purpose if this mostly a copy-paste from the regular syringes, exactly?

Reply:
Because I need to have them function like syringes, but unable to manually inject chemicals or draw blood.